src/Tools/Source/RunTests/HelixTestRunner.cs

```diff
@@ -0,0 +1,448 @@
+            // uploads. These settings limit the dump file size + produce a .json detailing crash 
+            // reasons that work better with Helix size limitations.
+            if (isMac)
+            {
+                command.AppendLine($"{setEnvironmentVariable} DOTNET_DbgEnableMiniDump=1");
+                command.AppendLine($"{setEnvironmentVariable} DOTNET_DbgMiniDumpType=1");
+                command.AppendLine($"{setEnvironmentVariable} DOTNET_EnableCrashReport=1");
+            }
+
+            // Set the dump folder so that dotnet writes all dump files to this location automatically. 
+            // This saves the need to scan for all the different types of dump files later and copy
+            // them around.
+            var helixDumpFolder = isUnix
+                ? @"$HELIX_DUMP_FOLDER/crash.%d.%e.dmp"
+                : @"%HELIX_DUMP_FOLDER%\crash.%d.%e.dmp";
+            command.AppendLine($"{setEnvironmentVariable} DOTNET_DbgMiniDumpName=\"{helixDumpFolder}\"");
+
+            command.AppendLine(isUnix ? "env | sort" : "set");
+
+            // Create a payload directory that contains all the assemblies in the work item in separate folders.
+            var payloadDirectory = Path.Combine(artifactsDir, "bin");
+
+            // Update the assembly groups to test with the assembly paths in the context of the helix work item.
+            workItemInfo = workItemInfo with { Filters = workItemInfo.Filters.ToImmutableSortedDictionary(kvp => kvp.Key with { AssemblyPath = GetHelixRelativeAssemblyPath(kvp.Key.AssemblyPath) }, kvp => kvp.Value) };
+
+            AddRehydrateTestFoldersCommand(command, workItemInfo, isUnix);
+
+            // Build an rsp file to send to dotnet test that contains all the assemblies and tests to run.
+            // This gets around command line length limitations and avoids weird escaping issues.
+            // See https://docs.microsoft.com/en-us/dotnet/standard/commandline/syntax#response-files
+            var rspFileName = $"vstest_{workItemInfo.PartitionIndex}.rsp";
+            var xmlResultsFileName = $"workitem_{workItemInfo.PartitionIndex}.xml";
+            var rspFileContent = GetRspFileContent(workItemInfo, platform, xmlResultsFileName);
+            WriteAllTextAndLog(Path.Combine(payloadDirectory, rspFileName), rspFileContent, logsDir);
+
+            // Build the command to run the rsp file.
+            // dotnet test does not pass rsp files correctly the vs test console, so we have to manually invoke vs test console.
+            // See https://github.com/microsoft/vstest/issues/3513
+            // The dotnet sdk includes the vstest.console.dll executable in the sdk folder in the installed version, so we look it up using the
+            // DOTNET_ROOT environment variable set by helix.
+            if (isUnix)
+            {
+                // $ is a special character in msbuild so we replace it with %24 in the helix project.
```

Comment:
If you're re-writing this in a followup PR, I suspect that all these commands should be moved to a single script file that gets executed in the workitem - instead of listing out each command in the msbuild project file.  Avoids these weird escaping issues.

Reply:
Yep that is definitely the plan. It's hard right now because everything is outputing into the `artifacts` folder. Consider that the copy-dumps.sh file is generated per work item but always to the same location on disk. Basically we're overwriting it every single time. 

The next PR puts every job into a different directory and we can then write out per job scripts much easier.